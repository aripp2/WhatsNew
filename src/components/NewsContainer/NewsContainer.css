section {
  width: 75%;
  display: flex;
  flex-wrap: wrap;
}