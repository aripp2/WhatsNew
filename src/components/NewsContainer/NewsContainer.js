import React from 'react';
import NewsArticle from '../NewsArticle/NewsArticle';
import './NewsContainer.css';

const NewsContainer = ({ news }) => {
  const articles = news.map(article => {
        return <NewsArticle 
          key={article.id}
          image={article.img}
          headline={article.headline}
          description={article.description}
          url={article.url}
        />
      })
  return ( 
    <section>
      {articles}
    </section>
  )
}

export default NewsContainer;