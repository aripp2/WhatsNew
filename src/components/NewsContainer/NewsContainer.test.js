import React from 'react';
import { shallow } from 'enzyme';
import NewsContainer from './NewsContainer';

describe('NewsContainer', () => {
  it('should match the snapshot with all data passed in correctly', () => {

    const articles = [{
        image: "url",
        headline: "This is the headline",
        description: "This is the description",
        url:"link" 
      },
      {
        image: "another url",
        headline: "another headline",
        description: "another description",
        url: "another link" 
      }];
  
    const wrapper = shallow(<NewsContainer 
      news={articles}
    />);

    expect(wrapper).toMatchSnapshot();
  });
});


