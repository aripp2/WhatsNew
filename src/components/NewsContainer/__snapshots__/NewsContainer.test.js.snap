// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewsContainer should match the snapshot with all data passed in correctly 1`] = `
<section>
  <NewsArticle
    description="This is the description"
    headline="This is the headline"
    url="link"
  />
  <NewsArticle
    description="another description"
    headline="another headline"
    url="another link"
  />
</section>
`;
